### PR TITLE
docs(messaging): add step-by-step instructions for image notification

### PR DESCRIPTION
**Pull request by davidepalazzo**
### Description
Based on #4218 this new piece of documentation adds a guide to setup a Notification Service Extension. 

It has screenshots to make the process a bit more visual. I wasn't too sure if you guys have a specific process for hosting images so I uploaded it on Gifyu for now. Let me know whether that needs changing or not.

Thanks!

### Related issues
#4218

### Checklist

- I read the [Contributor Guide](../CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
  - [x] Yes
- My change supports the following platforms;
  - [ ] `Android`
  - [x] `iOS`
- My change includes tests;
  - [ ] `e2e` tests added or updated in `packages/\*\*/e2e`
  - [ ] `jest` tests added or updated in `packages/\*\*/__tests__`
- [ ] I have updated TypeScript types that are affected by my change.
- This is a breaking change;
  - [ ] Yes
  - [x] No

### Test Plan

---

Think `react-native-firebase` is great? Please consider supporting the project with any of the below:

- 👉 Star this repo on GitHub ⭐️
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter
🔥

### CODE REVIEW

**Comment by Pittan** [comment]
Thank you for creating this awesome document!  

Suggestion:  
Maybe you're missing to modify this file :pray: 
https://github.com/invertase/react-native-firebase/blob/master/docs/sidebar.yaml

**Comment by davidepalazzo** [comment]
Thanks for that @Pittan, I totally missed it. Added!

**Comment by mikehardy** [comment]
I had meetings all day today so couldn't make many individual contributions, hoping to review this (and merge it, I also hope) tomorrow. Thanks for putting this together @davidepalazzo

File: docs/sidebar.yaml [hunk 1]
```diff
@@ -62,6 +62,7 @@
       - '/messaging/ios-permissions'
     - - Notifications
       - '/messaging/notifications'
+      - '/messaging/notifications-with-image'
```
Comment (mikehardy) [inline docs/sidebar.yaml:65]:
```suggestion
    - - Notifications with Images
      - '/messaging/notifications-with-image'
```

**Comment by mikehardy** [comment]
I think the sidebar entries need a header and then the file to link, I just committed a change that should do that? We'll see when the preview is done building from the vercel bot

I asked the Invertase folks how to do images as they all seem to be hosted on prismic.io right now and I don't know how I would get images there - thanks for putting them on gifyu at least temporarily though, it helps to see them in the preview

**Comment by mikehardy** [comment]
Needs attention from either @Salakar or @Ehesp to share knowledge on how to get images hosted correctly, as this new page contains Xcode explanation images

**Comment by mikehardy** [comment]
Yep - sidebar is good now in the preview, just waiting on guidance for image hosting and this is good to go!

**Comment by davidepalazzo** [comment]
Thanks for adding the title @mikehardy

**Comment by Salakar** [comment]
We should move the images into the website directory to be sourced from there as assets (PRs welcome to set this up on the project 🤞 ). Happy for these to be on gifyu for now though (though might be good to have copies of them in an assets folder in the website directory or something for safe keeping).

Could we also change the title, description and sidebar name to be more explicit that it's for iOS - like the iOS permissions one is.

File: docs/messaging/notifications-with-image.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
+---
+title: Notifications with Image
```
Comment (mikehardy) [inline docs/messaging/notifications-with-image.md:2]:
```suggestion
title: iOS Notification Images
```

File: docs/messaging/notifications-with-image.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+description: Displaying an image in a notification.
```
Comment (mikehardy) [inline docs/messaging/notifications-with-image.md:3]:
```suggestion
description: Displaying an image in an iOS notification.
```

File: docs/sidebar.yaml [hunk 4]
```diff
@@ -62,6 +62,8 @@
       - '/messaging/ios-permissions'
     - - Notifications
       - '/messaging/notifications'
+    - - Notifications with Images
```
Comment (mikehardy) [inline docs/sidebar.yaml:65]:
```suggestion
    - - iOS Notification Images
```

**Review by mikehardy — COMMENTED** [review]
Labelling page as iOS specific

**Review by mikehardy — CHANGES_REQUESTED** [review]
good feedback and sure - I just made it 'iOS Notification Images' per your request
@davidepalazzo could you make an assets directory and put the source images in there?
At that point - using the previews from Vercel that are auto-published on changes - you might be able to actually link them up directly, can you try that?

If that doesn't work we will at least have the assets for safe-keeping as requested and can figure out how to migrate all images as a separate issue to unblock this one

**Comment by davidepalazzo** [comment]
@mikehardy done.

I created an `assets` folder and placed it inside a new `static` folder. The `static` directory by default gets copied into the `public` folder which is reachable from the base URL. I've done this off the top of my head based on what I remembered about [Gatsby](https://www.gatsbyjs.com/docs/static-folder/#adding-assets-outside-of-the-module-system).

Another way of doing it, would be [enabling images support in markdown using a Gatsby plugin](https://www.gatsbyjs.com/docs/working-with-images-in-markdown/#using-the-mdx-plugin). It's straight forward but maybe material for a different branch.

@Salakar if you think it could be of use in the long run then I'm happy to create a new PR when I have some time in the coming days.

I tried to add some logic to the folder structure and files naming. It mirrors the docs — this should keep things easy and organised for future contributors that want to add screengrabs.
```js
static
├── docs
│   ├── {package-name} // e.g. messaging
│   │    └─── {page-name}-{image-description} // e.g. notification-installation-screenshot.png
```

**Review by mikehardy — APPROVED** [review]
That looks great to me, glad you knew some gatsby off the top of your head, I never would have known how to link those up! I'll leave any plugin stuff to a separate PR I think this is good to go and helps people now

**Comment by mikehardy** [comment]
Fantastic stuff, thank you @davidepalazzo !

**Comment by davidepalazzo** [comment]
No problem @mikehardy, glad we were able to sort it out!

**Pull request merged**